### PR TITLE
Enable hash_to_field

### LINKED ISSUE #490: Enable hash_to_field
# Problem

Trying to use `hash_to_field` in the std_lib will hit this [panic](https://github.com/noir-lang/noir/blob/55ed79d2a3ed350bfd167eadf9bf35784dc38e80/crates/noirc_evaluator/src/ssa/integer.rs#L494).

# Solution

I added `OPCODE::HashToField` to the match statement where I return the max size of the result type of the instruction (which is the native field). This is not enough it seems though as I can produce failing proofs. If you look at PR #489 you can recreate this by trying to prove and verify the newly added `hash_to_field` test under `nargo/tests/test_data`. 

# Alternatives considered

`hash_to_field` uses `blake2s` under the hood so I could convert my field to a byte array and then use blake2s. However, it is much easier to work with the native Field type and `hash_to_field` lets you avoid this conversion inconvenience. 

# Additional context

`nargo prove` will output "linear_eval is not 0." This is an error from barretenberg when constructing the proof. This points to something being wrong with how we are generating the `hash_to_field` gate.

### PR DESCRIPTION
**Pull request by vezenovm**
# Related issue(s)

Resolves #490 

# Description

This PR aims to allow usage of `hash_to_field` in Noir. This allows for convenient `blake2s` hashing without having to convert a Field to a byte array. 

## Summary of changes

I simply added a line to return the max size of the return type as it was missing for `OPCODE::HashToField`. This allows us to build and prove a circuit that uses `hash_to_field`. However, this change alone resulted in being able to generate proofs that can return false. More context be found in issue #490. We know this is an error when as we see "linear_eval is not 0." is printed on `nargo prove`. The fix is outline in the below section. 

## Dependency additions / changes

The `aztec_backend` commit used was updated to reflect a one line change when we create `hash_to_field` constraints in the C++ backend. 

In the `aztec_backend` when setting up the constraint inputs and outputs we use `assignment.fetch_nearest_bytes` which reverses the byte array returned by `to_bytes`. The method `fetch_nearest_bytes` can be found [here](https://github.com/noir-lang/noir/blob/dff8f393706c102055e78b4d97df207d73640eee/crates/noir_field/src/generic_ark.rs#L257). While its usage in the `aztec_backend` acvm_interop can be found [here](https://github.com/noir-lang/aztec_backend/blob/e3d4c7974dee07f17c9768f0e3f35e84ead175d3/barretenberg_static_lib/src/acvm_interop/pwg/gadget_call.rs#L124). When creating the hash to field constraints in C++ we treat the field inputs the same as u8 inputs, however, inside of `fetch_nearest_bytes` we are reversing the byte array we are planning to hash. This is fine to use for u8 inputs (which we do for foreign blake2s OPCODE) but for larger datatypes such as Fields we are reversing the endianness. This PR simply follows what we do in the Rust acvm_interop so that the inputs to the hash function used in Rust and the C++ match. However, a refactor could be done to make endianness more explicit throughout.

The relevant `aztec_backend` PR can be found here: https://github.com/noir-lang/aztec_backend/pull/27

The `aztec-connect/barretenberg` PR can be found here: https://github.com/noir-lang/aztec-connect/pull/6

## Test additions / changes

I added a basic test that returns the digest of `hash_to_field` on a singular Field input. 

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

# Additional context

(If applicable.)

### CODE REVIEW

**Review by guipublic — DISMISSED** [review]

**Comment by vezenovm** [comment]
@guipublic I see you approved, but this produces false proofs at the moment. I mention it in the issue linked and PR description.

File: crates/nargo/tests/test_data/hash_to_field/src/main.nr [hunk 1]
```diff
@@ -0,0 +1,6 @@
+use dep::std;
+
+fn main(input : Field) -> pub Field {
+    let digest = std::hash::hash_to_field([input]);
+    digest
```
Comment (jfecher) [inline crates/nargo/tests/test_data/hash_to_field/src/main.nr:5]: The let statement is unnecessary here

**Review by jfecher — COMMENTED** [review]
Looks good. I assume the issue with it creating false proofs has been fixed?

**Comment by vezenovm** [comment]
> Looks good. I assume the issue with it creating false proofs has been fixed?

Yeah it has. It was fixed in this [commit](https://github.com/noir-lang/aztec-connect/commit/6a810068375c90c00e2ca795823ff43ee8ebd79b). This change has been reflected on the master branch of `aztec_backend` and the commit here is the most recent `aztec_backend` commit.

File: crates/nargo/tests/test_data/hash_to_field/src/main.nr (hunk 1 above)
Comment (vezenovm) [reply crates/nargo/tests/test_data/hash_to_field/src/main.nr]: Removed

**Review by jfecher — APPROVED** [review]
👍

**Comment by vezenovm** [comment]
It turns out the most recent commit for `aztec_backend` has a couple errors where we are redownloading the CRS every time we use `nargo prove` or `nargo verify` when using `nargo` built from the `barretenberg_static_lib`. I don't this functionality to be merged into master so I am holding off on merging this PR just yet. I just merged this fix (https://github.com/noir-lang/aztec_backend/pull/33) that an external dev submitted. I am having some issues on my local machine with a corrupted CRS still. So once I am sure everything is working, I will merge this PR.

**Comment by vezenovm** [comment]
Ok everything looks to be working after this fix (https://github.com/noir-lang/aztec_backend/pull/33). The commit reference in this PR now references the most recent commit in `aztec_backend`

**Review by jfecher — APPROVED** [review]

**Pull request merged**